File: an2_sem4/MPP/Homework/lab2/cs/FestivalSellpoint/FestivalSellpoint.Repo/BiletDbRepo.cs
using FestivalSellpoint.Domain;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FestivalSellpoint.Repo
{
    public class BiletDbRepo : DatabaseRepoUtils<int, Bilet>, IBiletRepo
    {
        ISpectacolRepo SpectacolRepo;

        public BiletDbRepo(IDictionary<string, string> props, ISpectacolRepo spectacolRepo) : base(props)
        {
            SpectacolRepo = spectacolRepo;
        }

        public void Add(Bilet e)
        {
            int result = ExecuteNonQuery("insert into \"Bilet\" (\"numeCumparator\", \"spectacol\", \"nrLocuri\") " +
                "values (@numeC, @spectacol, @nrLocuri)", new Dictionary<string, object>
                {
                    { "@numeC", e.NumeCumparator },
                    { "@spectacol", e.Spectacol.Id},
                    { "@nrLocuri", e.NrLocuri},
                });
            if (result == 0)
                throw new RepositoryException("Bilet was not added");
        }

        public IEnumerable<Bilet> GetAll()
        {
            return Select("select * from \"Bilet\"");
        }

        public Bilet GetById(int id)
        {
            return SelectFirst("select * from \"Bilet\" where \"id\"=@id", new Dictionary<string, object>
            {
                { "@id", id },
            });
        }

        public IEnumerable<Bilet> getBySpectacol(Spectacol spectacol)
        {
            return Select("select * from \"Bilet\" where \"spectacol\"=@spectacol", new Dictionary<string, object>
            {
                { "@spectacol", spectacol.Id },
            });
        }

        public void Remove(int id)
        {
            throw new NotImplementedException("Cannot remove Bilet");
        }

        public void Update(Bilet e)
        {
            throw new NotImplementedException("Cannot update Bilet");
        }

        protected override Bilet DecodeReader(IDataReader reader)
        {
            var numeC = reader["numeCumparator"] as string;
            var spectId = Convert.ToInt32(reader["spectacol"]);
            var nrLocuri = Convert.ToInt32(reader["nrLocuri"]);
            var id = Convert.ToInt32(reader["id"]);

            var b = new Bilet(numeC, nrLocuri, SpectacolRepo.GetById(spectId));
            b.Id = id;
            return b;
        }
    }
}

File: an2_sem4/MPP/Homework/sapt11/java/repo/src/main/java/ro/ubbcluj/cs/stefnmubb/festivalsellpoint/repo/BiletDbRepo.java
package ro.ubbcluj.cs.stefnmubb.festivalsellpoint.repo;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Bilet;
import ro.ubbcluj.cs.stefnmubb.festivalsellpoint.domain.Spectacol;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;
import java.util.stream.StreamSupport;

public class BiletDbRepo extends DatabaseRepoUtils<Integer, Bilet>
    implements  IBiletRepo {

    private static final Logger logger= LogManager.getLogger(BiletDbRepo.class);

    ISpectacolRepo spectacolRepo;
    public BiletDbRepo(Properties props, ISpectacolRepo spectacolRepo) {
        super(props);
        this.spectacolRepo = spectacolRepo;
    }

    @Override
    public Bilet decodeResult(ResultSet result) throws SQLException, EntityRepoException {
        var id = result.getInt("id");
        var numeCumparator = result.getString("numeCumparator");
        var nrLocuri = result.getInt("nrLocuri");
        var spectacolId = result.getInt("spectacol");
        var spectacol = spectacolRepo.getById(spectacolId);
        var bilet = new Bilet(numeCumparator, nrLocuri, spectacol);
        bilet.setId(id);
        return bilet;
    }

    @Override
    public Iterable<Bilet> getBySpectacol(Spectacol spectacol) throws EntityRepoException {
        return select("select * from \"Bilet\" where \"spectacol\"=?", spectacol);
    }

    @Override
    public void add(Bilet bilet) throws EntityRepoException {
        logger.trace("Inserting {}", bilet);

        executeNonQuery("insert into \"Bilet\" (\"numeCumparator\", \"nrLocuri\", \"spectacol\") values (?, ?, ?)",
                bilet.getNumeCumparator(),
                bilet.getNrLocuri(),
                bilet.getSpectacol().getId());

        logger.info("Inserted successfully");
        logger.traceExit();
    }

    @Override
    public void update(Bilet bilet) throws EntityRepoException {
        logger.trace("Updating bilet {}", bilet);
        executeNonQuery("update \"Bilet\" set " +
                "\"numeCumparator\"=?," +
                "\"nrLocuri\"=?," +
                "\"spectacol\"=? where \"id\"=?",
                bilet.getNumeCumparator(),
                bilet.getNrLocuri(),
                bilet.getSpectacol().getId(),
                bilet.getId());
        logger.info("Updated bilet {}", bilet);
        logger.traceExit();
    }

    @Override
    public void remove(Integer id) throws EntityRepoException {
        logger.trace("Removing bilet id={}", id);
        executeNonQuery("delete from \"Bilet\" where id = ?", id);
        logger.info("Removed bilet id={}", id);
        logger.traceExit();
    }

    @Override
    public Bilet getById(Integer id) throws EntityRepoException {
        return selectFirst("select * from \"Bilet\" where \"id\"=?",id);
    }

    @Override
    public Iterable<Bilet> getAll() throws EntityRepoException {
        return select("select * from \"Bilet\"");
    }
}
